src/gantt.ts

```diff
@@ -1659,12 +1664,38 @@ module powerbi.extensibility.visual {
             taskSelection: UpdateSelection<Task>,
             taskConfigHeight: number): void {
 
+            const groupTasks: boolean = this.viewModel.settings.general.groupTasks;
+            let newLabelPosition: ResourceLabelPositions | null = null;
+            if (groupTasks && !Gantt.groupTasksPrevValue) {
+                newLabelPosition = ResourceLabelPositions.Top;
+            }
+
+            if (!groupTasks && Gantt.groupTasksPrevValue) {
+                newLabelPosition = ResourceLabelPositions.Right;
+            }
+
+            if (newLabelPosition) {
+                this.host.persistProperties(<VisualObjectInstancesToPersist>{
+                    merge: [{
+                        objectName: "taskResource",
+                        selector: null,
+                        properties: { position: newLabelPosition }
+                    }]
+                });
+
+                this.viewModel.settings.taskResource.position = newLabelPosition;
+                newLabelPosition = null;
+            }
+
+            Gantt.groupTasksPrevValue = groupTasks;
```

Comment:
You shouldn't change the static variable. Please use the variable as variable of instance